plugin.rb

```diff
@@ -1,65 +1,35 @@
 # frozen_string_literal: true
 
-# name: discourse-plugin-office365-auth
-# about: Enable Login via Office365
-# version: 0.0.1
+# name: discourse-microsoft-auth
+# about: Enable Login via Microsoft Identity Platform
+# version: 1.0
 # authors: Matthew Wilkin
-# url: https://github.com/discourse/discourse-plugin-office365-auth
+# url: https://github.com/discourse/discourse-microsoft-auth
 
-require 'auth/oauth2_authenticator'
-require File.expand_path('../omniauth-microsoft365.rb', __FILE__)
+require_relative "lib/omniauth-microsoft365"
 
-enabled_site_setting :office365_enabled
+enabled_site_setting :microsoft_auth_enabled
 
-class Office365Authenticator < ::Auth::OAuth2Authenticator
-  PLUGIN_NAME = 'oauth-office365'
+register_svg_icon "fab-microsoft"
 
+class ::MicrosoftAuthenticator < ::Auth::ManagedAuthenticator
   def name
     'microsoft_office365'
```

Comment:
Why is the authenticator still called `microsoft_office365`?

Reply:
As above, we can't change it without a lot of pain